benchconnect/README.md

```diff
@@ -18,6 +18,28 @@ With [pipx](https://pypa.github.io/pipx/):
 pipx install benchconnect@git+https://github.com/conbench/conbench.git@main#subdirectory=benchconnect
 ```
 
+## Environment variables
+
+benchconnect relies on a number of environment variables to obtain various
+metadata:
+
+- `CONBENCH_URL`: Required. The URL of the Conbench server without a trailing
+slash, e.g. `https://conbench.example.com`
+- `CONBENCH_EMAIL`: The email to use for Conbench login. Only required if the
+server is private.
+- `CONBENCH_PASSWORD`: The password to use for Conbench login. Only required
+if the server is private.
+- `CONBENCH_PROJECT_REPOSITORY`: The repository name (in the format `org/repo`) or the
```

Comment:
I was asking before about GitHub-specificity because that's really a question that I think users might have here, and the answer to that question helps with naming.

Must this be a repository that exists in GitHub? 

What is the implication of setting `org/repo` that does not exist in GitHub?

If we want to keep these parameters GitHub-specific, then I believe it makes a lot of sense to pull `GITHUB` into the names. If this discussion is already too annoying and long by now feel free to ignore :).

Reply:
Right now, there is a baked-in assumption everywhere in this repo that any project tracked by Conbench exists in GitHub. If it doesn't exist in GitHub, pretty much any feature using commit history is turned off, which reduces the usefulness of Conbench. 

It would probably take a lot of work to adjust Conbench code to accommodate for repositories hosted elsewhere. However, if we did, we'd probably still need the concepts of `CONBENCH_PROJECT_REPOSITORY` etc, and it would be a breaking change to remove the `GITHUB_` string out of these env vars. So I'm okay with leaving that string out in the first place.

Reply:
> It would probably take a lot of work to adjust Conbench code to accommodate for repositories hosted elsewhere.

Though we don't need to support this right now (so we should continue to defer it until it's actually necessary), I'm not sure this would be _a lot_ of work to adjust. Ideally the connections into github in the backend are isolated enough that it would be "just" writing new methods to connect to the gitlab or bitbucket or wherever APIs and plop the git relations into the database all the same, and then some `s/"github.com"/config.git_host_setting/g` for URLs. Of course that's not trivial, but it shouldn't be a lot of work (and if it is, that might be a sign that our factorization of our current github integration isn't ideal).

Reply:
Not anything we need to change now (and certainly not in this PR), but something to think about as we continue to maintain this codebase